mlrun/builder.py

```diff
@@ -229,7 +229,21 @@ def make_kaniko_pod(
         if end == -1:
             end = len(dest)
         repo = dest[dest.find("/") + 1 : end]
-        configure_kaniko_ecr_init_container(kpod, registry, repo)
+
+        # if no secret is given, assume ec2 instance has attached role which provides read/write access to ECR
+        assume_instance_role = not config.httpdb.builder.docker_registry_secret
+        configure_kaniko_ecr_init_container(kpod, registry, repo, assume_instance_role)
+
+        # project secret might conflict with the attached instance role
+        # ensure "AWS_ACCESS_KEY_ID", "AWS_SECRET_ACCESS_KEY" have no values or else kaniko will fail
+        # due to credentials conflict / lack of permission on given credentials
+        if assume_instance_role:
```

Comment:
I wonder what will happen if the user provides the env vars and we do not assume role which means we mount a docker registry secret (which is actually the AWS creds).
maybe we should block these env vars in all flows. I wonder if it's even possible to use them instead of mounting the docker registry secret like we do and like the kaniko docs say.

Reply:
that i believe wont matter because you mount the relevant secrets to init container/container *after* the project secrets so those take precedence by nature